lib/frame_enum.ml

```diff
@@ -1,67 +1,63 @@
 open Core_kernel.Std
 
-module type E = sig
+module type Enumerated = sig
   type t [@@deriving enumerate]
 end
 
-module type Enum = sig
-  include E
+module type Enumerable = sig
+  include Enumerated
 
   val to_enum : t -> int
   val of_enum : int -> t option
   val max : int
   val min : int
 end
 
-module type L = sig
-  include E
-  val list : (int * t) list
+module type Tabulated = sig
+  include Enumerated
+  val alli : (int * t) list
 end
 
-module Make_list(A : E) = struct
+module Tabulate(A : Enumerated) : Tabulated with type t = A.t = struct
   include A
-  let list = List.mapi all ~f:(fun i x -> i, x)
+  let alli = List.mapi all ~f:(fun i x -> i, x)
 end
 
-module Make_enum(A : L) : Enum with type t := A.t = struct
+module Enumerate(A : Tabulated) : Enumerable with type t := A.t = struct
   include A
 
-  let sorted = List.sort ~cmp:(fun (i,_) (j,_) -> compare i j) list
-
   let to_enum t =
-    fst @@ List.find_exn ~f:(fun (_,x) -> x = t) list
+    fst @@ List.find_exn ~f:(fun (_,x) -> x = t) alli
 
   let of_enum i =
-    List.find ~f:(fun (x,_) -> x = i) list |>
+    List.find ~f:(fun (x,_) -> x = i) alli |>
```

Comment:
You're still using the polymorphic comparison. If you need it, then put it as a requirement.